app/StateToken.php

```diff
@@ -24,6 +27,73 @@ class StateToken extends Model
     protected $hidden = [
     ];
 
+    /**
+     * StateToken constructor. Always end up with a UUID
+     * @param array $attributes
+     */
+    public function __construct(array $attributes = [])
+    {
+        // If empty or used, replace with a uuid
+        parent::__construct($attributes);
+
+        // if it exists and is used
+        if (array_key_exists('uuid', $attributes)) {
+            if (self::isUsedToken($attributes['uuid'])) {
+                $this->uuid = self::generateUnusedToken();
+            }
+        } else {
+            // If it doesn't exist, make one too.
+            $this->uuid = self::generateUnusedToken();
+        }
+    }
+
+    /**
+     * Makes and checks for an unused token
+     *
+     * @return string
+    */
+    public static function generateUnusedToken()
+    {
+        do {
+            // TODO: Deal with possibility uuid4() may throw an exception of it's own?
+            try {
+                $candidate = Uuid::uuid4()->toString();
+            } catch (\Exception $e) {
+                // Uuid4() throws exceptions, apparently! Log that and die, I guess?
+                Log::warning($e->getMessage());
+                abort(500, $e->getMessage());
+            }
+            // Check if it's in use.
+            $usedToken = self::isUsedToken($candidate);
+        } while ($usedToken === true);
+
+        return $candidate;
+    }
+
+    /**
+     * Tidies Tokens of a certain age.
+     * @param int $age
+     * @return bool|null
+     */
+    public static function tidy($age = 30)
+    {
+        $expireDate = Carbon::today()->subDays($age)->format('Y-m-d H:i:s');
+        return self::where('created_at', '<', $expireDate)->delete();
+    }
+
+    /**
+     * Checks a UUID has been used
+     * @param $candidate
+     * @return bool
+     */
+    public static function isUsedToken($candidate)
```

Comment:
This would be free if uuid was constrained to 'unique' but I am probably missing something obvious.

Reply:
I think uuid v4 is random in the sense that the chance of a conflict is 1 in [`2^122, or 5.3x10^36 (5.3 undecillion) `](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)). Neil is being _very_ cautious here :)